src/components/createTestCase/CreateTestCase.tsx

```diff
@@ -47,28 +52,47 @@ const Alert = styled.div<AlertProps>(({ status = "default" }) => [
 const CreateTestCase = () => {
   const navigate = useNavigate();
   const { id } = useParams<keyof navigationParams>() as navigationParams;
-  const testCaseService = useTestCaseServiceApi();
+  // Avoid infinite dependency render. May require additional error handling for timeouts.
+  const testCaseService = useRef(useTestCaseServiceApi());
   const [alert, setAlert] = useState<AlertProps>(null);
   const { measureId } = useParams<{ measureId: string }>();
   const [testCase, setTestCase] = useState<TestCase>(null);
   const formik = useFormik({
     initialValues: {
+      title: "",
       description: "",
     } as TestCase,
     validationSchema: Yup.object().shape({
+      title: Yup.string(),
       description: Yup.string(),
     }),
     onSubmit: async (values: TestCase) => await handleSubmit(values),
   });
+  const { resetForm } = formik;
 
   useEffect(() => {
-    if (!testCase && id) {
-      testCaseService.getTestCase(id, measureId).then((tc: TestCase) => {
-        setTestCase(tc);
-        formik.resetForm({ values: tc });
-      });
+    if (id) {
+      const updateTestCase = () => {
```

Comment:
@mcmcphillips Is there an advantage to wrapping this logic in a function instead of inlining?

Reply:
@jkotanchik-SB 
I don't believe there is any advantage aside from code readability